pkg/volume/flexvolume/attacher.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+type flexVolumeAttacher struct {
+	driverName string
+	execPath   string
+	host       volume.VolumeHost
+
+	driverSupportsGetDeviceMountPath *bool
+}
+
+var _ volume.Attacher = &flexVolumeAttacher{}
+
+// Attach is part of the volume.Attacher interface
+func (a *flexVolumeAttacher) Attach(spec *volume.Spec, hostName string) error {
+	call := NewDriverCall(a.execPath, attachCmd, exec.New())
+	call.AppendSpec(spec, nil, a.host)
+	call.Append(hostName)
+
+	_, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		return (*attacherDefaults)(a).Attach(spec, hostName)
+	}
+	return err
+}
+
+// WaitForAttach is part of the volume.Attacher interface
+func (a *flexVolumeAttacher) WaitForAttach(spec *volume.Spec, timeout time.Duration) (string, error) {
+	call := NewDriverCallWithTimeout(a.execPath, waitForAttachCmd, exec.New(), timeout)
+	call.AppendSpec(spec, nil, a.host)
+
+	status, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		return (*attacherDefaults)(a).WaitForAttach(spec, timeout)
+	} else if err != nil {
+		return "", err
+	}
+	return status.Device, nil
+}
+
+// GetDeviceMountPath is part of the volume.Attacher interface
+func (a *flexVolumeAttacher) GetDeviceMountPath(spec *volume.Spec) string {
+	if !*a.driverSupportsGetDeviceMountPath {
+		return (*attacherDefaults)(a).GetDeviceMountPath(spec)
+	}
+	call := NewDriverCall(a.execPath, getDeviceMountPathCmd, exec.New())
+	call.AppendSpec(spec, nil, a.host)
+
+	status, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		*a.driverSupportsGetDeviceMountPath = false
+		return (*attacherDefaults)(a).GetDeviceMountPath(spec)
+	} else if err != nil {
+		glog.Errorf("FlexVolume: driver bug in %s: returned an error on getdevicemountpath: %s", a.execPath, err)
+		return ""
+	}
+	return status.Path
+}
+
+// MountDevice is part of the volume.Attacher interface
+func (a *flexVolumeAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMountPath string, mounter mount.Interface) error {
+	call := NewDriverCall(a.execPath, mountDeviceCmd, exec.New())
+	call.AppendSpec(spec, nil, a.host)
```

Comment:
Fix the order to mountdevicecmd? current format `mount <mount dir> <mount device> <json params>`